Assets/Scripts/Spawner.cs

```diff
@@ -0,0 +1,59 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UniRx;
+using System;
+using Lean.Pool;
+
+public class Spawner : MonoBehaviour
+{
+    private Collider SpawnArea;
+    private IDisposable SpawnDisposable;
+
+    public GameObject[] RingPrefabs;
```

Comment:
in here if it doesn't have to be accessible from other classes variables should be private with [SerializeField] before them so we can edit the value in inspecter while keeping the data safe and encapsulated